astropy/io/ascii/qdp.py

```diff
@@ -0,0 +1,578 @@
+                    try:
+                        values.append(int(v))
+                    except ValueError:
+                        values.append(float(v))
+            current_rows.append(values)
+            continue
+
+        if datatype == "new":
+            # Save table to table_list and reset
+            if current_rows is not None:
+                new_table = Table(names=colnames, rows=current_rows)
+                new_table.meta["initial_comments"] = initial_comments.strip().split("\n")
+                new_table.meta["comments"] = comment_text.strip().split("\n")
+                # Reset comments
+                comment_text = ""
+                table_list.append(new_table)
+                current_rows = None
+            continue
+
+    # At the very end, if there is still a table being written, let's save
+    # it to the table_list
+    if current_rows is not None:
+        new_table = Table(names=colnames, rows=current_rows)
+        new_table.meta["initial_comments"] = initial_comments.strip().split("\n")
+        new_table.meta["comments"] = comment_text.strip().split("\n")
+        table_list.append(new_table)
+
+    return table_list
+
+
+def _understand_err_col(colnames):
+    """Get which column names are error columns
+
+    Examples
+    --------
+    >>> colnames = ['a', 'a_err', 'b', 'b_perr', 'b_nerr']
+    >>> serr, terr = _understand_err_col(colnames)
+    >>> np.allclose(serr, [1])
+    True
+    >>> np.allclose(terr, [2])
+    True
+    >>> serr, terr = _understand_err_col(['a', 'a_nerr'])
+    Traceback (most recent call last):
+    ...
+    ValueError: Missing positive error...
+    >>> serr, terr = _understand_err_col(['a', 'a_perr'])
+    Traceback (most recent call last):
+    ...
+    ValueError: Missing negative error...
+    """
+    shift = 0
+    serr = []
+    terr = []
+
+    for i, col in enumerate(colnames):
+        if col.endswith("_err"):
+            # The previous column, but they're numbered from 1!
+            # Plus, take shift into account
+            serr.append(i - shift)
+            shift += 1
+        elif col.endswith("_perr"):
+            terr.append(i - shift)
+            if len(colnames) == i + 1 or not colnames[i + 1].endswith('_nerr'):
+                raise ValueError("Missing negative error")
+            shift += 2
+        elif col.endswith("_nerr") and not colnames[i - 1].endswith('_perr'):
+            raise ValueError("Missing positive error")
+    return serr, terr
+
+
+def _read_table_qdp(qdp_file, names=None, table_id=None, delimiter=None):
+    """Read a table from a QDP file
+
+    Parameters
+    ----------
+    qdp_file : str
+        Input QDP file name
+
+    Other parameters
+    ----------------
+    table_id : int, default 0
+        Number of the table to be read from the QDP file. This is useful
+        when multiple tables present in the file. By default, the first is read.
+
+    names : list of strings
```

Comment:
Docstring parameters should have the same order as in the function. Also, `list of str`